### PR TITLE
[tibber] Enhancement, added channel and channel corrections due to API changes

### LINKED ISSUE #9796: [tibber] Unable to Initalize binding without a subscription
## Expected Behavior
The binding is supposed to work with or without a subscription

## Current Behavior
Apparently, Tibber has switched from using emtpy arrays to null values for subscription queries without subscription account. Thus the initialisation fails, and you never get live data measurements in OH.

## Possible Solution
Need to make changes to the code wrt initalisation to ignore null values rather than empty arrays

## Steps to Reproduce (for Bugs)
N/A

## Context
Tibber binding can't be initialized without a Subscription

## Your Environment
Openhabian
Linux ZZZZZZZ 5.4.83-v7+ #1379 SMP Mon Dec 14 13:08:57 GMT 2020 armv7l GNU/Linux
```
262 │ Active │  80 │ 3.0.0                   │ openHAB Add-ons :: Bundles :: Tibber Binding
264 │ Active │  80 │ 2.8.5                   │ Gson
```

### PR DESCRIPTION
**Pull request by kjoglum**
Binding adjustments/enhancement, added channel and channel corrections due to API changes.
[https://community.openhab.org/t/tibber-binding/67019/120?u=kjoglums](https://community.openhab.org/t/tibber-binding/67019/120?u=kjoglums) 

Closes #9796

Signed-off-by: Stian Kjoglum <stiankj@online.no>

### CODE REVIEW

File: bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java [hunk 1]
```diff
@@ -163,13 +164,14 @@ public void getURLInput(String url) throws IOException {
...
                     updateState(CURRENT_STARTSAT, new DateTimeType(timestamp));
+                    updateState(CURRENT_LEVEL, new StringType(myObject.get("level").toString()));
 
                 } catch (JsonSyntaxException e) {
                     updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
                             "Error communicating with Tibber API: " + e.getMessage());
                 }
             }
-            if (jsonResponse.contains("daily")) {
+            if (jsonResponse.contains("daily") && !jsonResponse.contains("[]") && !jsonResponse.contains("\"daily\":null")) {
```
Comment (cpmeister) [inline bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java:174]: I feel that `!jsonResponse.contains("[]")` might be too broad of a check with regards to checking the entire response. You should make sure to narrow down the check to just the `daily` field.

File: bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java [hunk 2]
```diff
@@ -188,7 +190,7 @@ public void getURLInput(String url) throws IOException {
                             "Error communicating with Tibber API: " + e.getMessage());
                 }
             }
-            if (jsonResponse.contains("hourly")) {
+            if (jsonResponse.contains("hourly") && !jsonResponse.contains("[]") && !jsonResponse.contains("\"hourly\":null")) {
```
Comment (cpmeister) [inline bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java:193]: same suggestion here.

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java (hunk 1 above)
Comment (kjoglum) [reply bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java]: Good point, narrowed down to `"daily":{"nodes":[]`

File: bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java (hunk 2 above)
Comment (kjoglum) [reply bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java]: Good point, narrowed down to `"hourly":{"nodes":[]`

**Review by cpmeister — APPROVED** [review]
LGTM

**Pull request merged**